azurelinuxagent/common/cgroupapi.py

```diff
@@ -606,27 +627,45 @@ def _is_systemd_failure(scope_name, process_output):
 
     def start_extension_command(self, extension_name, command, timeout, shell, cwd, env, stdout, stderr,
                                 error_code=ExtensionErrorCodes.PluginUnknownFailure):
-        scope_name = "{0}_{1}".format(self._get_extension_cgroup_name(extension_name), uuid.uuid4())
+        scope = "{0}_{1}".format(self._get_extension_cgroup_name(extension_name), uuid.uuid4())
 
         process = subprocess.Popen(
-            "systemd-run --unit={0} --scope {1}".format(scope_name, command),
+            "systemd-run --unit={0} --scope {1}".format(scope, command),
             shell=shell,
             cwd=cwd,
             stdout=stdout,
             stderr=stderr,
             env=env,
             preexec_fn=os.setsid)
 
-        logger.info("Started extension using scope '{0}'", scope_name)
-        extension_cgroups = []
+        scope_name = scope + '.scope'
 
-        def create_cgroup(controller):
-            cgroup_path = os.path.join(CGROUPS_FILE_SYSTEM_ROOT, controller, 'system.slice', scope_name + ".scope")
-            extension_cgroups.append(CGroup.create(cgroup_path, controller, extension_name))
+        logger.info("Started extension in unit '{0}'", scope_name)
 
-        self._foreach_controller(create_cgroup, 'Cannot create cgroup for extension {0}; '
-                                                'resource usage will not be tracked.'.format(extension_name))
-        self.track_cgroups(extension_cgroups)
+        try:
+            # systemd-run creates the scope under the system slice by default
+            cgroup_relative_path = os.path.join('system.slice', scope_name)
+
+            cpu_cgroup_mountpoint, memory_cgroup_mountpoint = self.get_cgroup_mount_points()
+
+            if cpu_cgroup_mountpoint is None:
+                logger.info("The CPU controller is not mounted; will not track resource usage")
+            else:
+                cpu_cgroup_path = os.path.join(cpu_cgroup_mountpoint, cgroup_relative_path)
```

Comment:
Just curious, would this path and the path from `get_process_cgroup_paths` be same?

Reply:
Yes. I was planning on using that function, but then I realize I do not have the process of the extension (I do have the process id of systemd-run, which the agent calls, but not of the extension itself.)

The documentation of systemd-run specifies that the cgroup is created under the system slice (unless changed by the --slice argument) so this should be safe.

Reply:
Ahh ok makes sense. 
I have another question (possibly outside the scope of this PR), why dont we create a new slice for each extension when invoking them? Using the `--slice` argument that you mentioned? Or were we actually doing that at some point but reverted to the system slice for some unknown issues?

Reply:
Yes, we already have the code that creates the slices and starts extensions in them, but it is not enabled because some versions of systemd simply ignore the --slice argument. As I start working on extensions I will add code to use those features when they are available.